File: Pearson-Interview/app/src/main/java/raghu/me/myapplication/ListAdapter.java
package raghu.me.myapplication;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import raghu.me.myapplication.model.Users;

import java.util.ArrayList;
import java.util.List;

public class ListAdapter extends RecyclerView.Adapter<ListAdapter.MyViewHolder> {

    private List<Users> mList = new ArrayList<>();
    private OnClickListener onClickListener;
    public ListAdapter(Context context) {
        onClickListener = (OnClickListener) context;
    }

    public interface OnClickListener {
        public void onClick(Users user);
    }
    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        View v = (View) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.row_list, parent, false);

        MyViewHolder vh = new MyViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.name.setText(mList.get(position).getName());
        holder.emailId.setText(mList.get(position).getEmail());
        holder.rootView.setTag(position);
        holder.rootView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int pos = (Integer) v.getTag();
                onClickListener.onClick(mList.get(pos));
            }
        });
    }



    @Override
    public int getItemCount() {
        return mList.size();
    }

    public void setData(List<Users> list) {
        mList.addAll(list);
        notifyDataSetChanged();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public TextView name,emailId;
        private View rootView;
        public MyViewHolder(View v) {
            super(v);
            name = v.findViewById(R.id.name);
            emailId = v.findViewById(R.id.email);
            rootView = v.findViewById(R.id.rootLayout);


        }
    }


}

File: Pearson-Interview/app/src/main/java/raghu/me/myapplication/ListAdapter.kt
package raghu.me.myapplication

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import raghu.me.myapplication.ListAdapter.MyViewHolder
import raghu.me.myapplication.model.Users

import java.util.ArrayList

class ListAdapter(context: Context) : RecyclerView.Adapter<ListAdapter.MyViewHolder>() {

    private val mList = ArrayList<Users>()
    private val onClickListener: OnClickListener

    init {
        onClickListener = context as OnClickListener
    }

    interface OnClickListener {
        fun onClick(user: Users)
    }

    override fun onCreateViewHolder(parent: ViewGroup, i: Int): MyViewHolder {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.row_list, parent, false) as View

        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.name.text = mList[position].name
        holder.emailId.text = mList[position].email
        holder.rootView.tag = position
        holder.rootView.setOnClickListener { v ->
            val pos = v.tag as Int
            onClickListener.onClick(mList[pos])
        }
    }


    override fun getItemCount(): Int {
        return mList.size
    }

    fun setData(list: List<Users>) {
        mList.addAll(list)
        notifyDataSetChanged()
    }

    class MyViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        // each data item is just a string in this case
        var name: TextView
        var emailId: TextView
        val rootView: View

        init {
            name = v.findViewById(R.id.name)
            emailId = v.findViewById(R.id.email)
            rootView = v.findViewById(R.id.rootLayout)


        }
    }


}
